apps/frontend/src/hooks/aave/useAaveBorrow.tsx

```diff
@@ -55,5 +55,28 @@ export const useAaveBorrow = () => {
     [setIsOpen, setTitle, setTransactions, aaveBorrowTransactionsFactory],
   );
 
-  return { handleBorrow };
+  const handleSwapBorrowRateMode = useCallback(
+    async (
+      asset: AssetDetailsData,
+      currentRateMode: BorrowRateMode,
+      opts?: TransactionFactoryOptions,
+    ) => {
+      if (!aaveBorrowTransactionsFactory) {
+        return;
+      }
+
+      setTransactions(
+        await aaveBorrowTransactionsFactory.swapBorrowRateMode(
```

Comment:
how are you handling errors here ?

Reply:
Actually it's a good point, I'm not doing any. There shouldn't be any errors arising from user behavior. But I'll add these checks in https://github.com/wakeuplabs-io/sovryn-dapp/pull/33